core/services/workflows/syncer/workflow_registry.go

```diff
@@ -2,39 +2,366 @@ package syncer
-	return nil
+// WithTicker allows external callers to provide a ticker to the workflowRegistry.  This is useful
+// for overriding the default tick interval.
+func WithTicker(ticker <-chan struct{}) func(*workflowRegistry) {
+	return func(wr *workflowRegistry) {
+		wr.ticker = ticker
+	}
 }
 
-func (w *WorkflowRegistry) Close() error {
-	return nil
+// NewWorkflowRegistry returns a new workflowRegistry.
+// Only queries for WorkflowRegistryForceUpdateSecretsRequestedV1 events.
+func NewWorkflowRegistry(
+	lggr logger.Logger,
+	orm WorkflowRegistryDS,
+	reader ContractReader,
+	gateway FetcherFunc,
+	addr string,
+	opts ...func(*workflowRegistry),
+) *workflowRegistry {
+	wr := &workflowRegistry{
+		lggr:    lggr.Named(name),
+		orm:     orm,
+		reader:  reader,
+		gateway: gateway,
+		cfg: ContractEventPollerConfig{
+			ContractName:      ContractName,
+			ContractAddress:   addr,
+			ContractEventName: ContractEventName,
+			QueryCount:        20,
+			StartBlockNum:     0,
+		},
+		stopCh:   make(services.StopChan),
+		eventsCh: make(chan WorkflowRegistryEventResponse),
+	}
+	wr.handler = newEventHandler(wr.lggr, wr.orm, wr.gateway)
+	for _, opt := range opts {
+		opt(wr)
+	}
+	return wr
+}
+
+// Start starts the workflowRegistry.  It starts two goroutines, one for querying the contract
+// and one for handling the events.
+func (w *workflowRegistry) Start(_ context.Context) error {
+	return w.StartOnce(w.Name(), func() error {
+		ctx, _ := w.stopCh.NewCtx()
+
+		w.wg.Add(1)
+		go func() {
+			defer w.wg.Done()
+			w.queryLoop(ctx, w.cfg)
+		}()
+
+		w.wg.Add(1)
+		go func() {
+			defer w.wg.Done()
+			w.handlerLoop(ctx)
+		}()
+
+		return nil
+	})
 }
 
-func (w *WorkflowRegistry) Ready() error {
+func (w *workflowRegistry) Close() error {
+	return w.StopOnce(w.Name(), func() error {
+		close(w.stopCh)
+		w.wg.Wait()
+		return nil
+	})
+}
+
+func (w *workflowRegistry) Ready() error {
 	return nil
 }
 
-func (w *WorkflowRegistry) HealthReport() map[string]error {
+func (w *workflowRegistry) HealthReport() map[string]error {
 	return nil
 }
 
-func (w *WorkflowRegistry) Name() string {
-	return "WorkflowRegistrySyncer"
+func (w *workflowRegistry) Name() string {
+	return name
 }
 
-func (w *WorkflowRegistry) SecretsFor(workflowOwner, workflowName string) (map[string]string, error) {
-	// TODO: actually get this from the right place.
-	return map[string]string{}, nil
+func (w *workflowRegistry) SecretsFor(ctx context.Context, workflowOwner, workflowName string) (map[string]string, error) {
+	return w.orm.SecretsFor(ctx, workflowOwner, workflowName)
 }
 
-func NewWorkflowRegistry() *WorkflowRegistry {
-	return &WorkflowRegistry{}
+// handlerLoop handles the events that are emitted by the contract.
+func (w *workflowRegistry) handlerLoop(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case resp, open := <-w.eventsCh:
+			if !open {
+				return
+			}
+
+			if resp.Err != nil || resp.Event == nil {
+				continue
```

Comment:
You could log the err here; atm we would have no way of knowing that an error occurred